### PR TITLE
MCS-562:  Initial design of test overview

### PR DESCRIPTION
**Pull request by brianpippin**
Initial design, I assume the psychs are going to ask for more stats or something on this page.

### CODE REVIEW

**Comment by brianpippin** [comment]
This is ready for a final review so we can get this merged to prod.

**Review by rartiss55 — APPROVED** [review]
Code LGTM! I don't think anyone is looking this closely, but we may want to hide the score card for Eval 3 interactive since it'll all be zero'd. 

Writing this out to double check my understanding:
hitRate = plausible scenes they got correct / the total plausible scenes for a category (in this case, for a given performer + test type + metadata level)
falseAlarm = implausible scenes they got incorrect / the total implausible scenes for a category

And then when you see "-", the category was not applicable. That all sound right?

**Comment by brianpippin** [comment]
> Code LGTM! I don't think anyone is looking this closely, but we may want to hide the score card for Eval 3 interactive since it'll all be zero'd.
> 
> Writing this out to double check my understanding: hitRate = plausible scenes they got correct / the total plausible scenes for a category (in this case, for a given performer + test type + metadata level) falseAlarm = implausible scenes they got incorrect / the total implausible scenes for a category
> 
> And then when you see "-", the category was not applicable. That all sound right?

Yes Rachel that is correct about hitRate and falseAlarm as explained to me by the psychologists.  Some of the hypercubes are only plausible scenes so they didn't want it to say 0 scenes because they thought it wasn't meaningful, so I went with - instead.  Ran it by them on slack a day or two ago.

File: analysis-ui/src/components/TestOverview/scorecardTable.jsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import TableRow from "@material-ui/core/TableRow";
+
+const scorecardDataQueryName = "getScoreCardData";
+const getScorecardDataQuery = gql`
+    query getScoreCardData($eval: String!, $categoryType: String!, $performer: String!, $metadata: String!) {
+        getScoreCardData(eval: $eval, categoryType: $categoryType, performer: $performer, metadata: $metadata) 
+    }`;
+
+const scorecardFields = [
+    {"title": "HyperCubeId", "key": "hypercubeID"},
```
Comment (ThomasSchellenbergNextCentury) [inline analysis-ui/src/components/TestOverview/scorecardTable.jsx:18]: Minor nitpick, but I don't think you need to capitalize the C in [hypercube](https://en.wikipedia.org/wiki/Hypercube)

File: analysis-ui/src/components/TestOverview/hypercubeResultsTable.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    constructor(props) {
+        super(props);
+
+        this.downloadCSV = this.downloadCSV.bind(this);
+    }
+
+    downloadCSV(tableData, tableHeaders, tableTitle) {
+        const columnDelimiter = ',';
+        const rowDelimter = '\n';
```
Comment (ThomasSchellenbergNextCentury) [inline analysis-ui/src/components/TestOverview/hypercubeResultsTable.jsx:43]: Should these exist as global variables?

File: node-graphql/server.schema.js [hunk 3]
```diff
@@ -396,8 +398,14 @@ const mcsResolvers = {
             }
 
             let results = await getComplexResults();
+            const uniqueResults = results.reduce((unique, o) => {
+                if(!unique.some(obj => obj._id.equals(o._id))) {
+                  unique.push(o);
+                }
+                return unique;
+            },[]);
```
Comment (ThomasSchellenbergNextCentury) [inline node-graphql/server.schema.js:406]: Good use of reduce.

File: node-graphql/server.statsFunctions.js [hunk 4]
```diff
@@ -262,7 +262,155 @@ function getChartData(isPlausibility, isPercent, scoreStats, isWeighted, evalTyp
...
 }
 
+/****  Found this function at:  https://memory.psych.mun.ca/models/dprime/
+ *      InvNormApprox:  Pass the hit rate and false alarm rate, and this
+ *      	routine returns zHit and zFa.  d' = zHit - zFa.
+ *          Converted from a basic routine provided by:
+ *      Brophy, A. L. (1986).  Alternatives to a table of criterion 
+ *              values in signal detection theory.  Behavior Research 
+ *              Methods, Instruments, & Computers, 18, 285-286.
+ ****/
```
Comment (ThomasSchellenbergNextCentury) [inline node-graphql/server.statsFunctions.js:272]: Thanks for the references! :)

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]

File: analysis-ui/src/components/TestOverview/hypercubeResultsTable.jsx (hunk 2 above)
Comment (brianpippin) [reply analysis-ui/src/components/TestOverview/hypercubeResultsTable.jsx:43]: No only want them in this function.

**Pull request merged**